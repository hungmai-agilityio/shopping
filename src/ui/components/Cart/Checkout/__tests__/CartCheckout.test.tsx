import { fireEvent, render, screen } from '@testing-library/react';

// Components
import CartCheckout from '@/ui/components/Cart/Checkout';

describe('Cart checkout component', () => {
  beforeEach(() => {
    jest.fn().mockClear();
  });

  const defaultProps = {
    price: 160,
    quantity: 4,
    onCheckout: jest.fn()
  };
  it('render with default props', () => {
    render(<CartCheckout {...defaultProps} />);

    expect(screen.getByText('Total')).toBeInTheDocument();
  });

  it('should update coupon code with "handleChange"', () => {
    render(<CartCheckout {...defaultProps} />);

    const inputElement = screen.getByRole('textbox') as HTMLInputElement;
    fireEvent.change(inputElement, { target: { value: 'SAVE20' } });

    expect(inputElement.value).toBe('SAVE20');
  });

  it('should clear coupon code and call onCheckout when handleCheckout is called', () => {
    render(<CartCheckout {...defaultProps} />);

    const inputElement = screen.getByRole('textbox') as HTMLInputElement;
    fireEvent.change(inputElement, { target: { value: 'DISCOUNT50' } });
    const checkoutButton = screen.getByText('Checkout');
    fireEvent.click(checkoutButton);

    expect(inputElement.value).toBe('');
    expect(defaultProps.onCheckout).toHaveBeenCalledTimes(1);
  });

  it('should be render match to snapshot', () => {
    const { container } = render(<CartCheckout {...defaultProps} />);

    expect(container).toMatchSnapshot();
  });
});
