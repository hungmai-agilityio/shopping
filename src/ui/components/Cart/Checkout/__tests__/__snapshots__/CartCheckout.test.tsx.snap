// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Cart checkout component should be render match to snapshot 1`] = `
<div>
  <div
    class="className w-cart-md h-cart-md bg-gray-light p-6 text-lg"
  >
    <p
      class="font-bold text-black lg:text-lg text-sm mb-1"
    >
      Coupon Code
    </p>
    <div
      class="mb-3"
    >
      <div
        class="relative"
      >
        <input
          class="w-full mt-1.5 font-bold outline-none font-lg text-dark-gray px-2 focus:border focus:border-green-300 h-12"
          type="text"
          value=""
        />
      </div>
    </div>
    <div
      class="flex justify-end"
    >
      <button
        class="truncate md:h-12 h-8 md:text-lg md:w-btn-md w-28 text-xs bg-gradient-primary text-white font-bold cursor-not-allowed"
        disabled=""
      >
        Check
      </button>
    </div>
    <div
      class="mt-9 flex justify-between"
    >
      <div>
        <p
          class="font-bold text-black lg:text-lg text-sm mb-1"
        >
          Subtotal
        </p>
        <span
          class="font-thin text-gray-400 lg:text-lg text-sm mb-1"
        >
          4
           Product
        </span>
      </div>
      <p
        class="font-thin text-black lg:text-lg text-sm mb-1"
      >
        $ 
        160
         USD
      </p>
    </div>
    <hr
      class="border-t-2 border-dashed border-gray-300 my-6"
    />
    <div
      class="flex justify-between"
    >
      <p
        class="font-bold text-black lg:text-lg text-sm mb-1"
      >
        Total
      </p>
      <p
        class="font-thin text-black lg:text-lg text-sm mb-1"
      >
        $ 
        160
         USD
      </p>
    </div>
    <div
      class="flex justify-end mt-5"
    >
      <button
        class="truncate md:h-12 h-8 md:text-lg md:w-btn-md w-28 text-xs bg-gradient-primary text-white font-bold"
      >
        Checkout
      </button>
    </div>
  </div>
</div>
`;
